task/MapFit.js

```diff
@@ -2,108 +2,63 @@ const through = require('through2')
+const fit = function (cmd, osmExtract, src, dst) {
   process.stdout.write('fitting ' + src + '...\n')
-  const lastLog = []
-  let success = true
-  const p = new Promise((resolve) => {
-    const dcmd = `docker pull ${dataToolImage}; docker run --rm -e TCMALLOC_LARGE_ALLOC_REPORT_THRESHOLD=2147483648 -v ${hostDataDir}:/data --rm ${dataToolImage} ${cmd} ${osmExtract} +init=epsg:3067 /${src} /${dst}`
-    const fit = exec(dcmd, { maxBuffer: constants.BUFFER_SIZE })
 
-    const checkError = (data) => {
-      debug(data)
-      lastLog.push(data.toString())
-      if (lastLog.length > 20) {
-        lastLog.splice(0, 1)
-      }
-      const txt = data.toString()
-      if (txt.indexOf('Traceback') !== -1 || txt.indexOf('IOError') !== -1) {
-        success = false
-      }
-    }
-
-    fit.stdout.on('data', data => checkError(data))
+  const dcmd = `docker pull ${dataToolImage}; docker run --rm -e TCMALLOC_LARGE_ALLOC_REPORT_THRESHOLD=2147483648 -v ${dataDir}:/data --rm ${dataToolImage} ${cmd} ${osmExtract} +init=epsg:3067 /${src} /${dst}`
 
-    fit.stderr.on('data', data => checkError(data))
-
-    fit.on('exit', function (code) {
-      if (code === 0 && success === true) {
-        resolve(true)
-      } else {
-        const log = lastLog.join('')
-        postSlackMessage(`Running command ${cmd} on ${src} failed: ${log} :boom:`)
-        process.stdout.write(`Running command ${cmd} failed:\n`)
-        process.stdout.write(`${src} ${log}\n`)
-        resolve(false)
-      }
-    })
-  })
-  return p
+  try {
+    execSync(dcmd)
+    return true
+  } catch (e) {
+    postSlackMessage(`Running command ${cmd} on ${src} failed :boom:`)
+    return false
+  }
 }
 
 module.exports = {
-  /**
-   * returns promise
-   */
-  fitGTFS: function (osmExtract, src, dst) {
-    return run('gtfs_shape_mapfit/fit_gtfs.bash', osmExtract, src, dst)
-  },
-
-  fitGTFSTask: (configs) => {
-    return through.obj(function (file, encoding, callback) {
+  fitGTFSTask: gtfsMap => {
+    return through.obj((file, encoding, callback) => {
       const gtfsFile = file.history[file.history.length - 1]
       const fileName = gtfsFile.split('/').pop()
       const relativeFilename = path.relative(process.cwd(), gtfsFile)
       const id = fileName.substring(0, fileName.indexOf('-gtfs'))
-      const config = configs[id]
-      if (config === undefined) {
-        process.stdout.write(`${gtfsFile} Could not find config for Id:${id}, ignoring fit...\n`)
+      const source = gtfsMap[id]
+      if (!source) {
+        process.stdout.write(`${gtfsFile} Could not find source for Id:${id}, ignoring fit...\n`)
         callback(null, null)
         return
       }
-      const osmFile = `${dataDir}/ready/osm/finland.pbf`
-
-      if (config.fit === false) {
+      if (!source.fit) {
         process.stdout.write(gtfsFile + ' fit skipped\n')
         callback(null, file)
         return
       }
-
+      const osmFile = `${dataDir}/ready/osm/finland.pbf`
```

Comment:
Why is this hardcoded here?

Reply:
It is an old sin, which I did not bother to fix yet. Will be fixed.

Reply:
Hadcoded chaged to configured version. mapfit cannot handle 2 OSM files (finland + estonia) but this will change as we are updating the mapfit process.